### PR TITLE
media-video/obs-studio: 28.0.2, -qtx11extras

### PR DESCRIPTION
**Pull request by mrusme**
As per https://github.com/obsproject/obs-studio/pull/4358/files - this way obs-studio buildable with -X

Signed-off-by: マリウス <marius@xn--gckvb8fzb.com>

### CODE REVIEW

**Comment by Chiitoo** [comment]
May also need to add note about qt5 builds, which are only somewhat supported by upstream at this point...

Sorry I haven't had the time to look into this yet.  Hopefully soon!

@ionenwks may be interested too.

**Comment by a17r** [comment]
> May also need to add note about qt5 builds, which are only somewhat supported by upstream at this point...

Until there are known bugs, we should not push users into unmasking Qt6 packages yet.

**Comment by mrusme** [comment]
Instructions were unclear, got my head stuck in the microwave. :-) 

Is there something missing that I should add? If so, kindly point out/explain what, as I didn't quite get what the Qt5 *note* and Qt6 *unmasking* is about, sorry.

**Comment by Chiitoo** [comment]
> > May also need to add note about qt5 builds, which are only somewhat supported by upstream at this point...
> 
> Until there are known bugs, we should not push users into unmasking Qt6 packages yet.

Yeah, didn't mean we'd go that way.



> Instructions were unclear, got my head stuck in the microwave. :-)
> 
> Is there something missing that I should add? If so, kindly point out/explain what, as I didn't quite get what the Qt5 _note_ and Qt6 _unmasking_ is about, sorry.

Apologies, they weren't really instructions.  I was just thinking if we should have some kind of a message about upstream qt5 support about which they say: "partly support it for ancient Ubuntus, but we urge everyone to upgrade to a system with Qt6 libs"

That does not necessarily mean that any and all issues reported from a qt5 build would not be handled at all though.

Our qt6 packaging is still not quite ready to be unmasked, so making obs-studio qt6-only would require uses to unmask qt6 for it, which we'd like to avoid.

File: media-video/obs-studio/obs-studio-28.0.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+		-DDISABLE_DECKLINK=$(usex !decklink)
+		-DDISABLE_FREETYPE=$(usex !truetype)
+		-DDISABLE_JACK=$(usex !jack)
+		-DDISABLE_LIBFDK=$(usex !fdk)
+		-DENABLE_NEW_MPEGTS_OUTPUT=OFF
+		-DENABLE_PIPEWIRE=$(usex pipewire)
+		-DDISABLE_PULSEAUDIO=$(usex !pulseaudio)
+		-DDISABLE_SPEEXDSP=$(usex !speex)
+		-DDISABLE_V4L2=$(usex !v4l)
+		-DDISABLE_VLC=$(usex !vlc)
```
Comment (esibun) [inline media-video/obs-studio/obs-studio-28.0.2.ebuild:186]: A lot of these configuration flags have changed with 28 - I believe all of them are now ENABLE_*; see https://github.com/obsproject/obs-studio/commit/49e9d49943d501eb372dee1afa78919c3d8640d3

Comment (mrusme) [reply media-video/obs-studio/obs-studio-28.0.2.ebuild]: Thanks, updated!

**Comment by mrusme** [comment]
Seem to be getting the following error with the change from `DISABLE_` to `ENABLE_`:

```
CMake Error: install(EXPORT "libobsTargets" ...) includes target "libobs" which requires target "jansson" that is not in any export set.
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    LIBOBS_PREFER_IMAGEMAGICK
```

Do not merge, need to investigate.

**Comment by Chiitoo** [comment]
Had a little bit of time to continue my work on the 9999, and looking at the changes here, and I think it would be nice to add a USE="websocket" so that it can remain optional.

Also considering flags for 'vst' and 'qt6' but those can wait.  VST is no longer a submodule, and seems tiny in terms of installed size, and probably not noticeable on compile time either.

Here are the current differences between my 9999 and your 28.0.2 (mine needs more testing and checking and fixing of dependencies, but running out of time now and not sure when I have more, so I wanted to put this out here already):

``` diff
--- obs-studio-28.0.2.ebuild	2022-10-24 21:13:28.268063593 +0300
+++ obs-studio-9999.ebuild	2022-10-24 23:36:10.259360070 +0300
@@ -9,27 +9,28 @@
 
 inherit cmake lua-single python-single-r1 xdg
 
-OBS_BROWSER_COMMIT="d1fa35dcbc384136e9e883f2d6071b14fd8f9a65"
-OBS_WEBSOCKET_COMMIT="265899f76f88a5be74747308fff3d35347ce43c5"
+CEF_DIR="cef_binary_5060_linux64"
 ASIO_COMMIT="b73dc1d2c0ecb9452a87c26544d7f71e24342df6"
 JSON_COMMIT="a34e011e24beece3b69397a03fdc650546f052c3"
+OBS_BROWSER_COMMIT="b6e0888084ab623f0a73e8cb7ee5dc341e56fda1"
+OBS_WEBSOCKET_COMMIT="5716577019b1ccda01a12db2cba35a023082b7ad"
 QR_COMMIT="8518684c0f33d004fa93971be2c6a8eca3167d1e"
 WEBSOCKETPP_COMMIT="56123c87598f8b1dd471be83ca841ceae07f95ba"
 
-CEF_DIR="cef_binary_4638_linux64"
-
 if [[ ${PV} == 9999 ]]; then
 	inherit git-r3
 	EGIT_REPO_URI="https://github.com/obsproject/obs-studio.git"
-	EGIT_SUBMODULES=( plugins/obs-browser plugins/obs-websocket )
 else
-	SRC_URI="https://github.com/obsproject/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
-	SRC_URI+=" browser? ( https://github.com/obsproject/obs-browser/archive/${OBS_BROWSER_COMMIT}.tar.gz -> obs-browser-${OBS_BROWSER_COMMIT}.tar.gz )"
-	SRC_URI+=" https://github.com/obsproject/obs-websocket/archive/${OBS_WEBSOCKET_COMMIT}.tar.gz -> obs-websocket-${OBS_WEBSOCKET_COMMIT}.tar.gz "
-	SRC_URI+=" https://github.com/chriskohlhoff/asio/archive/${ASIO_COMMIT}.tar.gz -> asio-${ASIO_COMMIT}.tar.gz "
-	SRC_URI+=" https://github.com/nlohmann/json/archive/${JSON_COMMIT}.tar.gz -> json-${JSON_COMMIT}.tar.gz "
-	SRC_URI+=" https://github.com/nayuki/QR-Code-generator/archive/${QR_COMMIT}.tar.gz -> qr-${QR_COMMIT}.tar.gz "
-	SRC_URI+=" https://github.com/zaphoyd/websocketpp/archive/${WEBSOCKETPP_COMMIT}.tar.gz -> websocketpp-${WEBSOCKETPP_COMMIT}.tar.gz "
+	SRC_URI="
+		https://github.com/obsproject/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+		browser? ( https://github.com/obsproject/obs-browser/archive/${OBS_BROWSER_COMMIT}.tar.gz -> obs-browser-${OBS_BROWSER_COMMIT}.tar.gz )
+		websocket? (
+			https://github.com/chriskohlhoff/asio/archive/${ASIO_COMMIT}.tar.gz -> asio-${ASIO_COMMIT}.tar.gz
+			https://github.com/nayuki/QR-Code-generator/archive/${QR_COMMIT}.tar.gz -> qr-${QR_COMMIT}.tar.gz
+			https://github.com/nlohmann/json/archive/${JSON_COMMIT}.tar.gz -> json-${JSON_COMMIT}.tar.gz
+			https://github.com/obsproject/obs-websocket/archive/${OBS_WEBSOCKET_COMMIT}.tar.gz -> obs-websocket-${OBS_WEBSOCKET_COMMIT}.tar.gz
+			https://github.com/zaphoyd/websocketpp/archive/${WEBSOCKETPP_COMMIT}.tar.gz -> websocketpp-${WEBSOCKETPP_COMMIT}.tar.gz
+	)"
 	KEYWORDS="~amd64 ~ppc64 ~x86"
 fi
 SRC_URI+=" browser? ( https://cdn-fastly.obsproject.com/downloads/${CEF_DIR}.tar.bz2 )"
@@ -40,8 +41,8 @@
 LICENSE="GPL-2"
 SLOT="0"
 IUSE="
-	+alsa browser decklink fdk jack lua nvenc pipewire
-	pulseaudio python speex +ssl truetype v4l vlc wayland
+	+alsa browser decklink fdk jack lua nvenc pipewire pulseaudio
+	python speex +ssl truetype v4l vlc wayland websocket
 "
 REQUIRED_USE="
 	browser? ( || ( alsa pulseaudio ) )
@@ -57,12 +58,9 @@
 	dev-libs/glib:2
 	dev-libs/jansson:=
 	dev-qt/qtcore:5
-	dev-qt/qtdeclarative:5
 	dev-qt/qtgui:5[wayland?]
-	dev-qt/qtmultimedia:5
 	dev-qt/qtnetwork:5
 	dev-qt/qtquickcontrols:5
-	dev-qt/qtsql:5
 	dev-qt/qtsvg:5
 	dev-qt/qtwidgets:5
 	dev-qt/qtxml:5
@@ -73,18 +71,16 @@
 	sys-apps/dbus
 	sys-apps/pciutils
 	sys-libs/zlib:=
-	virtual/udev
 	x11-libs/libX11
 	x11-libs/libXcomposite
 	x11-libs/libXfixes
-	x11-libs/libXinerama
-	x11-libs/libXrandr
 	x11-libs/libxcb:=
 	alsa? ( media-libs/alsa-lib )
 	browser? (
-		app-accessibility/at-spi2-atk
-		app-accessibility/at-spi2-core:2
-		dev-libs/atk
+		|| (
+			>=app-accessibility/at-spi2-core-2.46.0:2
+			( app-accessibility/at-spi2-atk dev-libs/atk )
+		)
 		dev-libs/expat
 		dev-libs/glib
 		dev-libs/nspr
@@ -95,13 +91,17 @@
 		net-print/cups
 		x11-libs/cairo
 		x11-libs/libdrm
-		x11-libs/libXScrnSaver
 		x11-libs/libXcursor
 		x11-libs/libXdamage
 		x11-libs/libXext
 		x11-libs/libXi
+		x11-libs/libxkbcommon
+		x11-libs/libXrandr
 		x11-libs/libXrender
+		x11-libs/libXScrnSaver
+		x11-libs/libxshmfence
 		x11-libs/libXtst
+		x11-libs/pango
 	)
 	fdk? ( media-libs/fdk-aac:= )
 	jack? ( virtual/jack )
@@ -115,9 +115,15 @@
 		media-libs/fontconfig
 		media-libs/freetype
 	)
-	v4l? ( media-libs/libv4l )
+	v4l? (
+		media-libs/libv4l
+		virtual/udev
+	)
 	vlc? ( media-video/vlc:= )
-	wayland? ( dev-libs/wayland )
+	wayland? (
+		dev-libs/wayland
+		x11-libs/libxkbcommon
+	)
 "
 RDEPEND="${DEPEND}"
 
@@ -141,6 +147,15 @@
 	default
 
 	if [[ ${PV} == 9999 ]]; then
+		local EGIT_SUBMODULES=()
+		use browser && EGIT_SUBMODULES+=( plugins/obs-browser )
+		use websocket && EGIT_SUBMODULES+=(
+			plugins/obs-websocket
+			plugins/obs-websocket/deps/asio
+			plugins/obs-websocket/deps/json
+			plugins/obs-websocket/deps/qr
+			plugins/obs-websocket/deps/websocketpp
+		)
 		git-r3_src_unpack
 	else
 		if use browser; then
@@ -148,32 +163,45 @@
 			mv obs-browser-${OBS_BROWSER_COMMIT} ${P}/plugins/obs-browser || die
 		fi
 
-		rm -d ${P}/plugins/obs-websocket || die
-		mv obs-websocket-${OBS_WEBSOCKET_COMMIT} ${P}/plugins/obs-websocket || die
-
-		rm -d ${P}/plugins/obs-websocket/deps/asio || die
-		mv asio-${ASIO_COMMIT} ${P}/plugins/obs-websocket/deps/asio || die
-
-		rm -d ${P}/plugins/obs-websocket/deps/json || die
-		mv json-${JSON_COMMIT} ${P}/plugins/obs-websocket/deps/json || die
+		if use websocket; then
+			rm -d ${P}/plugins/obs-websocket || die
+			mv obs-websocket-${OBS_WEBSOCKET_COMMIT} ${P}/plugins/obs-websocket || die
+
+			rm -d ${P}/plugins/obs-websocket/deps/asio || die
+			mv asio-${ASIO_COMMIT} ${P}/plugins/obs-websocket/deps/asio || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/json || die
+			mv json-${JSON_COMMIT} ${P}/plugins/obs-websocket/deps/json || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/qr || die
+			mv QR-Code-generator-${QR_COMMIT} ${P}/plugins/obs-websocket/deps/qr || die
+	￼
+			rm -d ${P}/plugins/obs-websocket/deps/websocketpp || die
+			mv websocketpp-${WEBSOCKETPP_COMMIT} ${P}/plugins/obs-websocket/deps/websocketpp || die
+		fi
+	fi
+}
 
-		rm -d ${P}/plugins/obs-websocket/deps/qr || die
-		mv QR-Code-generator-${QR_COMMIT} ${P}/plugins/obs-websocket/deps/qr || die
+src_prepare() {
+	# Avoid errors if we don't have 'obs-browser' or 'obs-websocket'.
+	if ! use browser; then
+		sed -i 's/FATAL_ERROR "obs-browser/STATUS "obs-browser/' plugins/CMakeLists.txt || die
+	fi
 
-		rm -d ${P}/plugins/obs-websocket/deps/websocketpp || die
-		mv websocketpp-${WEBSOCKETPP_COMMIT} ${P}/plugins/obs-websocket/deps/websocketpp || die
+	if ! use websocket; then
+		sed -i 's/FATAL_ERROR "obs-websocket/STATUS "obs-websocket/' plugins/CMakeLists.txt || die
 	fi
+
+	cmake_src_prepare
 }
 
 src_configure() {
 	local libdir=$(get_libdir)
 	local mycmakeargs=(
 		$(usev browser -DCEF_ROOT_DIR=../${CEF_DIR})
-		-DBUILD_BROWSER=$(usex browser)
-		-DBUILD_VST=no
-		-DENABLE_WAYLAND=$(usex wayland)
-		-DENABLE_AJA=OFF
 		-DENABLE_ALSA=$(usex alsa)
+		-DENABLE_AJA=OFF
+		-DENABLE_BROWSER=$(usex browser)
 		-DENABLE_DECKLINK=$(usex decklink)
 		-DENABLE_FREETYPE=$(usex truetype)
 		-DENABLE_JACK=$(usex jack)
@@ -181,16 +209,15 @@
 		-DENABLE_NEW_MPEGTS_OUTPUT=OFF
 		-DENABLE_PIPEWIRE=$(usex pipewire)
 		-DENABLE_PULSEAUDIO=$(usex pulseaudio)
+		-DENABLE_RTMPS=$(usex ssl ON OFF)
 		-DENABLE_SPEEXDSP=$(usex speex)
 		-DENABLE_V4L2=$(usex v4l)
 		-DENABLE_VLC=$(usex vlc)
+		-DENABLE_VST=ON
+		-DENABLE_WAYLAND=$(usex wayland)
+		-DENABLE_WEBSOCKET=$(usex websocket)
 		-DOBS_MULTIARCH_SUFFIX=${libdir#lib}
 		-DUNIX_STRUCTURE=1
-		-DWITH_RTMPS=$(usex ssl)
-
-		# deprecated and currently cause issues
-		# https://github.com/obsproject/obs-studio/pull/4560#issuecomment-826345608
-		-DLIBOBS_PREFER_IMAGEMAGICK=no
 	)
 
 	if [[ ${PV} != 9999 ]]; then
@@ -203,10 +230,10 @@
 		mycmakeargs+=(
 			-DENABLE_SCRIPTING_LUA=$(usex lua)
 			-DENABLE_SCRIPTING_PYTHON=$(usex python)
-			-DENABLE_SCRIPTING=yes
+			-DENABLE_SCRIPTING=ON
 		)
 	else
-		mycmakeargs+=( -DENABLE_SCRIPTING=no )
+		mycmakeargs+=( -DENABLE_SCRIPTING=OFF )
 	fi
 
 	cmake_src_configure

```

**Comment by Chiitoo** [comment]
Greetings!

Since I had some time in my hands, and there has been yet another new release already, I went ahead and created https://github.com/gentoo/gentoo/pull/28094 with my changes merged with yours.

Thank you!

**Comment by ionenwks** [comment]
Thanks, closing to continue in other PR as mentioned in previous comment.

**Pull request closed without merging**